src/main/java/ro/redteam/taskmanagementsystem/services/UserServiceImpl.java

```diff
@@ -37,4 +40,51 @@ public UserDTO createUser(UserDTO userDTO) {
             throw new EmailExistsException("Invalid email");
         }
     }
+
+    @Override
+    public UserDTO getUserById(Long id) {
+        if(userRepository.existsById(id)){
+            return  objectMapper.convertValue(userRepository.findById(id), UserDTO.class);
+        }else{
+            throw new UserNotFoundException("User does not exist!");
+        }
+    }
+
+    @Override
+    public List<UserDTO> getAllUsers() {
+        List<User> users = userRepository.findAll();
+        return users.stream()
+                .map(user -> objectMapper.convertValue(user, UserDTO.class))
+                .toList();
+    }
+
+    @Override
+    public String deleteUserById(Long id) {
+        if(userRepository.existsById(id)){
+            userRepository.deleteById(id);
+            return  "User deleted successfully!";
+        }else{
+            throw new UserNotFoundException("User does not exist!");
+        }
+    }
+
+    @Override
+    public UserDTO updateUserById(Long id, UserDTO newUserDTO) {
+        if(userRepository.existsById(id)){
+            User user = userRepository.findById(id).get();
+            if (newUserDTO.getEmail().isEmpty() || newUserDTO.getFirstName().isEmpty() || newUserDTO.getLastName().isEmpty()) {
+                throw new EmptyInputException("Filed empty");
+            }try {
+                user.setFirstName(newUserDTO.getFirstName());
+                user.setLastName(newUserDTO.getLastName());
+                user.setEmail(newUserDTO.getEmail());
+                //user.setTaskId(newUserDTO.getTaskId()); ~?
+                return objectMapper.convertValue(userRepository.save(user), UserDTO.class);
+            } catch (DataIntegrityViolationException e) {
+                throw new EmailExistsException("Invalid email");
+            }
+        }else{
+            throw new UserNotFoundException("User does not exist!");
+        }
```

Comment:
should be used optional, and if the user is present, the save method will update only the fields that are not empty, the task id should not be changed, the task id is updated when the assignee of a task is changed.